### DISCUSSION TITLE
Opentracing integration with go-redis, but there is no way to get metadata of current config/host?

### DISCUSSION THREAD

**Discussion by jiekun** [redis/go-redis#2078] [Ideas]
Hi maintainers.

We are trying to integrate **go-redis** with tracing and metrics reports. Normally we collect:
- command name (`cmd.Name()`)
- command execution time (`after hook - before hook`)
- full command (self-impl, like `cmdString(cmd Cmder, val interface{})` but we just don't need `val`)

When we try to report the host name (like a domain or `IP:Port`, or a self-config identifier), it seems we could not extract the config info from `cmd`. 

We used to set a few metadata in context using `WithContext()` after init. However, we noticed that the `UniversalClient` did not include this Method. So I would like to discuss it here.

It's ok not requiring the impl of `WithContext` in `UniversalClient`, we can still use a specific client. However, is there any way to get the **metadata** (not only the `addr`, I mean other metadata, if possible) from `Hook` ?

**Comment by auula** [comment]
> Hi maintainers.
> 
> We are trying to integrate **go-redis** with tracing and metrics reports. Normally we collect:
> 
> * command name (`cmd.Name()`)
> * command execution time (`after hook - before hook`)
> * full command (self-impl, like `cmdString(cmd Cmder, val interface{})` but we just don't need `val`)
> 
> When we try to report the host name (like a domain or `IP:Port`, or a self-config identifier), it seems we could not extract the config info from `cmd`.
> 
> We used to set a few metadata in context using `WithContext()` after init. However, we noticed that the `UniversalClient` did not include this Method. So I would like to discuss it here.
> 
> It's ok not requiring the impl of `WithContext` in `UniversalClient`, we can still use a specific client. However, is there any way to get the **metadata** (not only the `addr`, I mean other metadata, if possible) from `Hook` ?

Good open source software isn't what you're asking questions about, you're asking questions that make you look ignorant.

**Reply by jiekun** [reply]
Thank you! Lesson learned from ya. 😄
https://github.com/redis/go-redis/discussions/2113
